imsim/sky_model.py

```diff
@@ -108,7 +113,103 @@ def __call__(self, x, y):
         value at the CCD center.
         """
         return (self.a*x + self.b*y + self.c)/self.sky_level_center
-
+    
+class CCD_Fringing:
+    """
+    Class generates normalized fringing map. 
+    
+    The function call operator returns the normlized fringing level
+    as a function of pixel coordinates relative to the value at the 
+    CCD center.
+    """
+    def __init__(self,img_wcs,c_wcs,seed,spatial_vary):
+        self.img_wcs = img_wcs
+        self.c_wcs = c_wcs
+        self.spatial_vary = spatial_vary
+        self.seed = seed
+        
+    def generate_heightfield(self,fractal_dimension=2.5, n=4096):
+        '''
+        # Use the spectral sythesis method to generate a heightfield
+        '''
+        H = 1 - (fractal_dimension - 2)
+        kpow = -(H + 1.0) / 1.2
+        
+        A = np.zeros((n, n), complex)
+
+        kvec = np.fft.fftfreq(n)
+        k0 = kvec[n // 64]
+        kx, ky = np.meshgrid(kvec, kvec, sparse=True, copy=False)
+        ksq = kx ** 2 + ky ** 2
+        m = ksq > 0
+        random_seed = self.seed
+        gen = np.random.RandomState(random_seed)
```

Comment:
Don't use `np.random.RandomState`.  They don't guarantee consistent values across numpy versions, so that means the simulated values won't necessarily be portable across different systems.  Always use GalSim random number generators for any random values in simulations.  In this case, I think we do want to use the same random seed for a given detector every time.  So we don't actually want to tie it to the RNG that other things in the config file are using.  (Otherwise, different exposures would get different fringing patterns.)  So using det_num for the seed is a good idea.  But this line should be
```
gen = galsim.BaseDeviate(random_seed).np
```
The `np` property returns a numpy-style generator that can be used just like a np.random.Generator() instance.